app/containers/MessageBox/index.js

```diff
@@ -706,30 +791,53 @@ class MessageBox extends Component {
 		if (item.username === 'all' || item.username === 'here') {
 			return this.renderFixedMentionItem(item);
 		}
+		let testID;
+		switch (trackingType) {
+			case MENTIONS_TRACKING_TYPE_EMOJIS:
+				testID = `mention-item-${ item.name || item }`;
+				break;
+			case MENTIONS_TRACKING_TYPE_COMMANDS:
+				testID = `mention-item-${ item.command || item }`;
+				break;
+			default:
+				testID = `mention-item-${ item.username || item }`;
+		}
+
 		return (
 			<TouchableOpacity
 				style={styles.mentionItem}
 				onPress={() => this.onPressMention(item)}
-				testID={`mention-item-${ trackingType === MENTIONS_TRACKING_TYPE_EMOJIS ? item.name || item : item.username || item.name }`}
+				testID={testID}
 			>
-				{trackingType === MENTIONS_TRACKING_TYPE_EMOJIS
-					? [
-						this.renderMentionEmoji(item),
-						<Text key='mention-item-name' style={styles.mentionText}>:{ item.name || item }:</Text>
-					]
-					: [
-						<Avatar
-							key='mention-item-avatar'
-							style={{ margin: 8 }}
-							text={item.username || item.name}
-							size={30}
-							type={item.username ? 'd' : 'c'}
-							baseUrl={baseUrl}
-							userId={user.id}
-							token={user.token}
-						/>,
-						<Text key='mention-item-name' style={styles.mentionText}>{ item.username || item.name }</Text>
-					]
+
+				{(() => {
+					switch (trackingType) {
```

Comment:
avoid switch case

Reply:
its better then if else can you suggest something else